### PR TITLE
[1.1.x] Support for Tronxy and Zonestar LCD

### PR DESCRIPTION
**Pull request by drdelaney**
Code borrowed and modified from erikkallen/Marlin_tronxy.
Changed the ANET_KEYPAD_LCD to zonestar, since it is the same device.
I kept a define for ANET_KEYPAD_LCD to enable ZONESTAR_LCD.

### CODE REVIEW

**Comment by drdelaney** [comment]
Related to #8759

File: Marlin/pins_SANGUINOLOLU_11.h [hunk 1]
```diff
@@ -212,8 +212,12 @@
...
   #endif // !DOGLCD
 
-  #define BTN_EN1               11
-  #define BTN_EN2               10
+  #if DISABLED(ZONESTAR_LCD)
+
+    #define BTN_EN1               11
+    #define BTN_EN2               10
+
+  #endif
```
Comment (fiveangle) [inline Marlin/pins_SANGUINOLOLU_11.h]: Don't key off the universe where this panel is disabled - that will throw warnings for every other panel that defines these differently.

File: Marlin/Conditionals_LCD.h [hunk 2]
```diff
@@ -56,6 +56,10 @@
     #define ENCODER_FEEDRATE_DEADZONE 2
     #define REVERSE_MENU_DIRECTION
 
+  #elif ENABLED(ANET_KEYPAD_LCD) // Leaving for previous config compatability
+
+    #define ZONESTAR_LCD
+
```
Comment (fiveangle) [inline Marlin/Conditionals_LCD.h]: Either define existing `ANET_KEYPAD_LCD ` if `ENABLED(ZONESTAR_LCD)` is defined [_least attractive option as it just adds config bloat for almost no benefit_] or just accept that you'll use the _ANET_KEYPAD_LCD_ as-is on the Tronxy [preferred option].

**Comment by jcook423** [comment]
Forgive me if it exists, but is there code for the Zonestar LCD on the _Zrib board_ as well? 

I know support for the Zrib board itself was introduced into Marlin, but I cant seem to locate the LCD is the config.h file.  Selecting "#define REPRAP_DISCOUNT_SMART_CONTROLLER" will get the display on, but none of the 5 buttons will work.

Any help would be greatly appreciated, thank  you!

File: Marlin/pins_SANGUINOLOLU_11.h (hunk 1 above)
Comment (drdelaney) [reply Marlin/pins_SANGUINOLOLU_11.h]: Understood.  I will rework this.

File: Marlin/Conditionals_LCD.h (hunk 2 above)
Comment (drdelaney) [reply Marlin/Conditionals_LCD.h]: It seems the 5 button LCD2004 displays are used on more then just the ANET and Tronxy boards.  And in theory, can be used on any board.  I am having a hard time finding verification that these are by zonestar (I replicated this code from someone else), but it seems the other LCD defines are either LCD manufacturer specific or generic entries.  Unless these are by ANET (mine says Tronxy on it), I feel this should have a different define.  Perhaps FIVEBUTTON_KEYPAD or something.
I am seeing references for this type of board via ANet, Tronxy, Zonestar and Hesine.

**Comment by drdelaney** [comment]
I am not familiar enough with how to get the build to pass.  Since this wont work for RAMPS boards, and the travis builder does not seem to have the Sanguino data.

**Comment by drdelaney** [comment]
@jcook423 I am not familiar enough with the Zrib board, nor do I have one to mess around with. I would not feel comfortable doing any work to make the LCD work on that right now.

**Comment by jcook423** [comment]
> @jcook423 I am not familiar enough with the Zrib board, nor do I have one to mess around with. I would not feel comfortable doing any work to make the LCD work on that right now.             

No stress, TY. I'll look up the pinouts of the two boards and see if I get it working. :)

File: Marlin/Configuration.h [hunk 3]
```diff
@@ -111,15 +111,15 @@
  *
  * :[2400, 9600, 19200, 38400, 57600, 115200, 250000, 500000, 1000000]
  */
-#define BAUDRATE 250000
+#define BAUDRATE 115200
```
Comment (thinkyhead) [inline Marlin/Configuration.h]: You no change default config. Change default config bad.

Comment (drdelaney) [reply Marlin/Configuration.h]: Noted.  Was not sure how this worked. Thanks for the review and fixing.

**Pull request merged**